Sources/Plasma/Apps/plClient/CMakeLists.txt

```diff
@@ -90,17 +89,93 @@ if(WIN32)
         win32/res/Dirt.ICO
         win32/res/headspin.ico
     )
+elseif(APPLE)
+    set(plClient_SOURCES ${plClient_SOURCES}
+        Mac-Cocoa/main.mm
+        Mac-Cocoa/PLSKeyboardEventMonitor.mm
+        Mac-Cocoa/PLSView.mm
+        Mac-Cocoa/PLSLoginWindowController.mm
+        Mac-Cocoa/PLSPatcherWindowController.mm
+        Mac-Cocoa/PLSPatcher.mm
+        Mac-Cocoa/PLSServerStatus.mm
+        Mac-Cocoa/StringTheory_NSString.mm
+    )
+    set(plClient_HEADERS ${plClient_HEADERS}
+        Mac-Cocoa/PLSKeyboardEventMonitor.h
+        Mac-Cocoa/PLSView.h
+        Mac-Cocoa/PLSLoginWindowController.h
+        Mac-Cocoa/PLSPatcherWindowController.h
+        Mac-Cocoa/PLSPatcher.h
+        Mac-Cocoa/PLSServerStatus.h
+        Mac-Cocoa/StringTheory_NSString.h
+    )
+    set(RESOURCES 
+        Mac-Cocoa/MainMenu.xib
+        Mac-Cocoa/PLSLoginWindowController.xib
+        Mac-Cocoa/PLSPatcherWindowController.xib
+        Mac-Cocoa/banner.png
+        Mac-Cocoa/banner@2x.png
+        win32/res/Dirt.ICO
+    )
+    
+    list(APPEND plClient_RESOURCES
+        win32/res/resource.h
+        win32/res/banner.bmp
+        win32/res/Dirt.ICO
+        win32/res/headspin.ico
+    )
 else()
     list(APPEND plClient_SOURCES
         main.cpp
     )
 endif()
 
-plasma_executable(plClient WIN32 CLIENT INSTALL_PDB
-    SOURCES
-        ${plClient_SOURCES} ${plClient_HEADERS}
-        ${plClient_TEXT} ${plClient_RESOURCES}
-)
+if(APPLE)
+    plasma_executable(plClient CLIENT INSTALL_PDB SOURCES ${plClient_SOURCES} ${plClient_HEADERS} ${RESOURCES})
+    set(MAIN_NIBFILE "MainMenu")
+    set_target_properties(plClient PROPERTIES
+        BUNDLE TRUE
+        MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_SOURCE_DIR}/Mac-Cocoa/Info.plist.in"
+        MACOSX_BUNDLE_GUI_IDENTIFIER org.openuru.plClient
```

Comment:
openuru? :(

Reply:
Yeah, this decision was made before I knew... everything.

This ID is a little tricky because it maybe should be changed for each server. It definitely should be changed as it's signed by different orgs. So OpenUru and H'uru should probably have different identifiers.

Reply:
We have the cmake values `PRODUCT_CORE_NAME`, `PRODUCT_LONG_NAME`, and `PRODUCT_SHORT_NAME` that could help us. It is currently the convention that each shard is using its own client; however, this is not actually a requirement. The shard name is currently a runtime option stored in server.ini, so you can point your client at any shard that you like. In the original variant of Uru Live, there were actually multiple shards that you could select from a drop down on the login dialog. So, I'm not sure that encoding the shard name here is the right choice. I think that client lineage makes more sense. In some cases that may be shard related. In other cases, not so much. The configurable CMake product strings are probably the best for this.

Reply:
That seems reasonable.

Another key thing this string does - this is used to key things like saved passwords (and thus is also tied to our code signing.) So different IDs will have different saved password pools.

Because this is linked to code signing - that means a code signing account has to own the identifier. So if OpenUru picked up this work - it might make sense for them to have their own identifier because that client might be Cyan signed.